### PR TITLE
Adding new ALI api version 2024-08-01-preview

### PR DESCRIPTION
**Pull request by 8Gitbrix**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by 8Gitbrix** [comment]
I need help fixing the swagger errors on this PR. I can't find any of the properties that are mentioned in Swagger ModelValidation errors. Can someone please help me fix these errors?

**Comment by ms-bogdan-rotaru** [comment]
> I need help fixing the swagger errors on this PR. I can't find any of the properties that are mentioned in Swagger ModelValidation errors. Can someone please help me fix these errors?

For the `XMS_EXAMPLE_NOTFOUND_ERROR` error(s) you need to make sure you're referencing an example by using the `x-ms-example` attribute.

`id`, `name`, `type` and `systemMetadata` are read-only properties.

**Comment by ms-bogdan-rotaru** [comment]
Also, please have a look on the `Swagger LintDiff` check

**Comment by 8Gitbrix** [comment]
@ms-bogdan-rotaru Yes but I didn't define those properties, in fact is unclear where those properties are defined and how I'm supposed to fix it... 

Unfortunately, the input resources, AzureLargeInstance and AzureLargeStorage instance contain some nested readonly properties. I've specified all of them and suppressed them in my readme.md. Yet, the errors are still popping up.
 
Is there a way to simply suppress all nested properties instead of tirelessly defining all of them? Or what's the path to suppress the remaining read only properties?

**Comment by gary-x-li** [comment]
As discussed, please remove the suppressions such as `READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST` and fix them.

**Comment by mentat9** [comment]
>   - code: TrackedResourcesMustHavePut

Can you provide a more complete explanation for this suppression? I think we discussed this in office hours and considered it actually as a bug fix, since the previous spec was missing a couple of APIs. The APIs that don't implement PUT are already published, this spec is just documenting them.

However, if this API version is still in private preview, you definitely should fix it before fully rolling it out. If this is truly a read-only API (doesn't support PUT/PATCH), then it should be modeled as a proxy resource, not tracked. Basically, ARM can't manage it as a tracked resource if PUTs aren't going through ARM.

---
Refers to: specification/azurelargeinstance/resource-manager/readme.md:110 in 5be72cc. [](commit_id = 5be72cc6f21b14a1edc30ae496524611799b35f1, deletion_comment = False)

File: specification/azurelargeinstance/AzureLargeInstance.Management/models.tsp [hunk 1]
```diff
@@ -292,20 +292,44 @@ model AzureLargeInstanceProperties {
   @doc("Specifies the network settings for the Azure Large Instance.")
   networkProfile?: NetworkProfile;
 
+  @removed(Versions.v2024_08_01_preview)
+  @renamedFrom(Versions.v2024_08_01_preview, "azureLargeInstanceId")
   @doc("Specifies the Azure Large Instance unique ID.")
   @visibility("read")
+  azureLargeInstanceIdOld?: string;
+
```
Comment (mentat9) [inline specification/azurelargeinstance/AzureLargeInstance.Management/models.tsp:300]: I'm not familiar with this pattern. What is this for?

Comment (8Gitbrix) [reply specification/azurelargeinstance/AzureLargeInstance.Management/models.tsp:300]: This is a [workaround](https://azure.github.io/typespec-azure/docs/next/troubleshoot/versioning_operation_template_changes) in typespec that's needed to update the visibility / decorator of a variable.

**Comment by 8Gitbrix** [comment]
> > * code: TrackedResourcesMustHavePut
> 
> Can you provide a more complete explanation for this suppression? I think we discussed this in office hours and considered it actually as a bug fix, since the previous spec was missing a couple of APIs. The APIs that don't implement PUT are already published, this spec is just documenting them.
> 
> However, if this API version is still in private preview, you definitely should fix it before fully rolling it out. If this is truly a read-only API (doesn't support PUT/PATCH), then it should be modeled as a proxy resource, not tracked. Basically, ARM can't manage it as a tracked resource if PUTs aren't going through ARM.
> 
> Refers to: specification/azurelargeinstance/resource-manager/readme.md:110 in [5be72cc](https://github.com/Azure/azure-rest-api-specs/commit/5be72cc6f21b14a1edc30ae496524611799b35f1). [](commit_id = [5be72cc](https://github.com/Azure/azure-rest-api-specs/commit/5be72cc6f21b14a1edc30ae496524611799b35f1), deletion_comment = False)

This API version is the first public preview version to publish the PUT methods. If I remove the suppressions, the linters will fail due to the previous specs not containing the put method

**Review by faizmicrosoft — APPROVED** [review]

**Review by mentat9 — APPROVED** [review]

**Pull request merged**